src/cli/commands/force/lightning/lwc/preview.ts

```diff
@@ -96,98 +103,245 @@ export default class Preview extends Setup {
+            const appConfig = PreviewUtils.getAppConfig(
+                configFileJson,
+                platform,
+                targetApp
+            );
+            if (appConfig === null || appConfig === undefined) {
+                const errMsg = util.format(
+                    messages.getMessage(
+                        'error:invalidConfigFile:missingAppConfigDescription'
+                    ),
+                    targetApp,
+                    platform
+                );
+                return Promise.reject(
+                    new SfdxError(
+                        util.format(
+                            messages.getMessage(
+                                'error:invalidConfigFile:genericDescription'
+                            ),
+                            errMsg
+                        ),
+                        'lwc-dev-mobile'
+                    )
+                );
+            }
         }
-        return promise;
+
+        return Promise.resolve();
     }
 
-    public launchIOS(): Promise<boolean> {
-        const simName = CommandLineUtils.resolveFlag(
+    public launchPreview(): Promise<boolean> {
+        const platform = this.flags.platform;
+
+        const defaultDeviceName = CommandLineUtils.platformFlagIsIOS(platform)
+            ? iOSConfig.defaultSimulatorName
+            : androidConfig.defaultEmulatorName;
+
+        const device = CommandLineUtils.resolveFlag(
             this.flags.target,
-            iOSConfig.defaultSimulatorName
+            defaultDeviceName
         );
 
         const targetApp = CommandLineUtils.resolveFlag(
             this.flags.targetapp,
             PreviewUtils.BROWSER_TARGET_APP
         );
 
-        const targetAppArguments = CommandLineUtils.resolveFlag(
-            this.flags.targetapparguments,
-            ''
-        );
-
         const projectDir = CommandLineUtils.resolveFlag(
             this.flags.projectdir,
             process.cwd()
         );
 
-        const componentName = this.flags.componentname;
+        const configFile = CommandLineUtils.resolveFlag(
+            this.flags.configfile,
+            ''
+        );
+
+        const component = this.flags.componentname;
 
-        const launcher = new IOSLauncher(simName);
+        let targetAppArguments: Map<string, string> = new Map();
+        let launchActivity: string = '';
+        if (PreviewUtils.isTargetingBrowser(targetApp) === false) {
+            const json = PreviewUtils.getConfigFileAsJson(configFile);
 
-        return launcher.launchPreview(
-            componentName,
-            projectDir,
-            targetApp,
-            targetAppArguments
-        );
+            targetAppArguments = PreviewUtils.getAppLaunchArguments(
+                json,
+                platform,
+                targetApp
+            );
+
+            launchActivity = PreviewUtils.getAppLaunchActivity(json, targetApp);
```

Comment:
We should probably conditionalize the retrieval of `launchActivity` to Android use cases. I'm not sure what the backing assumptions are in `PreviewUtils.getAppLaunchActivity()` for its presence across platforms.

Reply:
`PreviewUtils.getAppLaunchActivity()` will automatically look for `targetApp` under `android` in the config file.
In the next line I will conditionally pick a launcher and won't use `activity` in `launchiOS`. It is only used when calling `launchAndroid`

That said, I can move things around to retrieve `activity` conditionally too.